src/dotnet-sourcelink/Program.cs

```diff
@@ -62,77 +59,82 @@ private static string GetSourceLinkVersion()
+            var pathArg = new Argument<string>("path")
+            {
+                Description = "Path to an assembly or .pdb"
+            };
+            var authArg = new Option<string>("--auth", "-a")
             {
                 Description = "Authentication method"
             };
             authArg.AcceptOnlyFromAmong(AuthenticationMethod.Basic);
 
-            var userArg = new CliOption<string>("--user", "-u")
+            var authEncodingArg = new Option<Encoding>("--auth-encoding", "-e")
+            {
+                CustomParser = arg => Encoding.GetEncoding(arg.Tokens.Single().Value),
+                Description = "Encoding to use for authentication value"
+            };
+
+            var userArg = new Option<string>("--user", "-u")
             {
                 Description = "Username to use to authenticate",
                 Arity = ArgumentArity.ExactlyOne
             };
 
-            var passwordArg = new CliOption<string>("--password", "-p")
+            var passwordArg = new Option<string>("--password", "-p")
             {
                 Description = "Password to use to authenticate",
                 Arity = ArgumentArity.ExactlyOne
             };
 
-            var offlineArg = new CliOption<bool>("--offline")
+            var offlineArg = new Option<bool>("--offline")
             {
                 Description = "Offline mode - skip validation of sourcelink URL targets"
             };
 
-            var test = new CliCommand("test", "TODO")
+            var test = new Command("test", "TODO")
             {
-                new CliArgument<string>("path")
-                {
-                    Description = "Path to an assembly or .pdb"
-                },
+                pathArg,
                 authArg,
-                new CliOption<Encoding>("--auth-encoding", "-e")
-                {
-                    CustomParser = arg => Encoding.GetEncoding(arg.Tokens.Single().Value),
-                    Description = "Encoding to use for authentication value"
-                },
+                authEncodingArg,
                 userArg,
                 passwordArg,
                 offlineArg,
             };
-            test.Action = CommandHandler.Create<string, string?, Encoding?, string?, string?, bool, ParseResult>(TestAsync);
+
+            test.SetAction((parseResult, cancellationToken) =>
+            {
+                string path = parseResult.GetValue(pathArg)!;
+                string? authMethod = parseResult.GetValue(authArg);
+                Encoding? authEncoding = parseResult.GetValue(authEncodingArg);
+                string? user = parseResult.GetValue(userArg);
+                string? password = parseResult.GetValue(passwordArg);
+                bool offline = parseResult.GetValue(offlineArg);
+
+                return TestAsync(path, authMethod, authEncoding, user, password, offline, parseResult, cancellationToken);
```

Comment:
`CommandHandler.Create` was part of NamingConventionBinder which would use reflection just to achieve what I've written here